crypto/x509/by_file.c

```diff
@@ -121,34 +118,37 @@ int X509_load_cert_file_ex(X509_LOOKUP *ctx, const char *file, int type,
                     break;
                 } else {
                     ERR_clear_last_mark();
+                    if (count == 0)
+                        ERR_raise(ERR_LIB_X509, X509_R_NO_CERTIFICATE_FOUND);
                     goto err;
                 }
             }
             ERR_clear_last_mark();
-            i = X509_STORE_add_cert(ctx->store_ctx, x);
-            if (!i)
+            if (!X509_STORE_add_cert(ctx->store_ctx, x)) {
+                if (count == 0)
+                    ERR_raise(ERR_LIB_X509, X509_R_NO_CERTIFICATE_FOUND);
                 goto err;
+            }
             count++;
-            X509_free(x);
-            x = NULL;
         }
-        ret = count;
     } else if (type == X509_FILETYPE_ASN1) {
         if (d2i_X509_bio(in, &x) == NULL) {
             ERR_raise(ERR_LIB_X509, ERR_R_ASN1_LIB);
             goto err;
         }
-        i = X509_STORE_add_cert(ctx->store_ctx, x);
-        if (!i)
+        count = X509_STORE_add_cert(ctx->store_ctx, x);
+        if (count == 0) {
+            ERR_raise(ERR_LIB_X509, X509_R_NO_CERTIFICATE_FOUND);
```

Comment:
Again, I would not raise this error here as the reason is not that a cert was not found.